src/record/record-handler.js

```diff
@@ -106,6 +120,47 @@ RecordHandler.prototype.handle = function( socketWrapper, message ) {
 };
 
 /**
+ * Tries to retrieve the record from the cache or storage. If not found in either
+ * returns false, otherwise returns true
+ * @param   {SocketWrapper} socketWrapper the socket that send the request
+ * @param   {Object} message parsed and validated message
+ *
+ * @private
+ * @returns {void}
+ */
+RecordHandler.prototype._hasRecord = function( socketWrapper, message ) {
+	var recordName = message.data[ 0 ],
+		onComplete = function( record ) {
+			var hasRecord = record ? 'T' : 'F';
+			socketWrapper.sendMessage( C.TOPIC.RECORD, C.ACTIONS.HAS, [ recordName, hasRecord ] );
+		};
+
+	new RecordRequest( recordName, this._options, socketWrapper, onComplete.bind( this ) );
+};
+
+/**
+ * Sends the records data current data once loaded from the cache, and null otherwise
+ *
+ * @param {SocketWrapper} socketWrapper the socket that send the request
+ * @param   {Object} message parsed and validated message
+ * @private
+ * @returns {void}
+ */
+RecordHandler.prototype._snapshot = function( socketWrapper, message ) {
+	var recordName = message.data[ 0 ],
+		onComplete = function( record ) {
+			if( record ) {
+				this._sendRecord( recordName, record, socketWrapper );
+			} else {
+				//TODO: This needs to be improved
```

Comment:
Agreed. Maybe we should rather send an error than introducing a new concept of null for missing